MapboxDirections/MBVisualInstructionDetailComponent.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+
+#if os(iOS) || os(tvOS)
+    import UIKit
+#endif
+
+
+/**
+ :nodoc:
+ Contains necessary features to build a more rich visual experience for a `RouteStep`.
+ */
+@objc(MBVisualInstructionComponent)
+public class VisualInstructionComponent: NSObject, NSSecureCoding {
+    
+    /**
+     :nodoc:
+     Single part of a visual instruction.
+     */
+    public let text: String?
+    
+    /**
+     :nodoc:
+     Dictionary containing `UITraitCollection` for scales 1.0, 2.0 and 3.0. Each key's value is a `URL`.
```

Comment:
There should be two documentation comments, one within the `#if` block:

> A dictionary containing the URLs to image representations of this component.
> 
> The keys of this dictionary are trait collections specifying different display scales. The values of this dictionary are URLs to remote images that you should prefer over the `text` property once they are available. Use the URL that is best suited to the system’s current traits.

and the other within the `#else` block

> A dictionary containing the URLs to image representations of this component.
> 
> The keys of this dictionary are screen scale factors. The values of this dictionary are URLs to remote images that you should prefer over the `text` property once they are available. Use the URL that is best suited to the current screen’s scale factor.